internal/ansible/dispatcher/dispatcher_events.go

```diff
@@ -0,0 +1,155 @@
+package dispatcher
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/google/uuid"
+	"github.com/jakub-dzon/k4e-device-worker/internal/ansible/model/message"
+	pb "github.com/redhatinsights/yggdrasil/protocol"
+	"github.com/spatialcurrent/go-simple-serializer/pkg/json"
+)
+
+// Event-generating functions for the Playbook Dispatcher
+type EventList struct {
+	Events []message.AnsibleRunnerJobEventYaml
+}
+
+func NewEventList() *EventList {
+	return &EventList{Events: []message.AnsibleRunnerJobEventYaml{}}
+}
+
+//	TODO: Implement this function
+func (el EventList) AddEvent(event message.AnsibleRunnerJobEventYaml) {
```

Comment:
This function should be equivalent to the function in [rhc-worker-playbook](github.com/RedHatInsights/rhc-worker-playbook/blob/acd51a1cab05322e38d9ce68113af61ddcc7de54/rhc_worker_playbook/server.py#L109).

I can't understand where the "ansible_facts" dictionary is defined. 
The messages schema are described in the playbook-dispatcher [schema folder](https://github.com/RedHatInsights/playbook-dispatcher/tree/master/schema) section, if I'm not mistaken.

Reply:
@machacekondra can you help?

Reply:
For the moment I set `gather_facts` to true so i can try to mimic the behavior. 
Probably, there is something in the `insights-client` validation, but I still need to investigate it.

Now I know how to get the data structure which contains the `ansible_facts` (for reference `AnsiblePlaybookJSONResults` of the go-ansible library)

Do you agree with this approach?